src/main/java/com/aphyr/riemann/client/ChainPromise.java

```diff
@@ -92,6 +92,11 @@ public Object unsafeDeref(final long time,
   }
 
   @Override
+  public void setCallback(Callback<T> callback) {
+    throw new UnsupportedOperationException("Can't set a callback on a chained promise; set it on the underlying promise instead.");
```

Comment:
I have no idea whether this is right or not, but I couldn't figure out a way to attach a callback here that made any sense.